modules/client/src/internal/client/decoding.ts

```diff
@@ -152,18 +151,7 @@ export class ClientDecoding extends ClientCore implements IClientDecoding {
     }
     const result = daoInterface.decodeFunctionData("setMetadata", data);
     const bytes = hexToBytes(result[0]);
-    const cid = new TextDecoder().decode(bytes);
-    const ipfsRegex =
-      /^Qm([1-9A-HJ-NP-Za-km-z]{44,}|b[A-Za-z2-7]{58,}|B[A-Z2-7]{58,}|z[1-9A-HJ-NP-Za-km-z]{48,}|F[0-9A-F]{50,})$/;
-    if (!ipfsRegex.test(cid)) {
-      throw new Error("The metadata URL defined on the DAO is invalid");
-    }
-    try {
-      const stringMetadata = await this.ipfs.fetchString(cid);
-      return JSON.parse(stringMetadata);
-    } catch {
-      throw new Error("Error reading data from IPFS");
-    }
+    return new TextDecoder().decode(bytes);
```

Comment:
Having some CiD format validation may still be good

Reply:
I don't think so, one of the reasons for splitting this is to allow users to use another protocols outside `ipfs` to sore their metadata, if we add a `cid ` validation here we will not allow the user to store the metadata outside `ipfs`

Reply:
See above, by this I mean having some function to detect whether it is an IPFS link that we can `cat()`. In the future, the same building block will remain relevant alongside other potential alternatives